Services/CO.CDP.Person.WebApi/Api/ValidationOption2.cs

```diff
@@ -0,0 +1,165 @@
+using DotSwashbuckle.AspNetCore.SwaggerGen;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.OpenApi.Models;
+using System.ComponentModel.DataAnnotations;
+using FluentValidation;
+using FluentValidation.Results;
+
+namespace CO.CDP.Person.WebApi.Api.ValidationSpikeTwo
+{
+    public record Person
+    {
+        [Required(AllowEmptyStrings = true)] public required string Id { get; init; }
+        [Required(AllowEmptyStrings = true)] public required string Name { get; init; }
+        public required int Age { get; init; }
+        [EmailAddress] public required string Email { get; init; }
+    }
+
+    public record NewPerson
+    {
+        [Required(AllowEmptyStrings = true)] public required string Name { get; init; }
+        public required int Age { get; init; }
+        [EmailAddress] public required string Email { get; init; }
+    }
+
+    public record UpdatedPerson
+    {
+        [Required(AllowEmptyStrings = true)] public required string Name { get; init; }
+        public required int Age { get; init; }
+        [EmailAddress] public required string Email { get; init; }
+    }
+
+    public static class EndpointExtensions
+    {
+        private static Dictionary<string, Person> _persons = Enumerable.Range(1, 5)
+            .ToDictionary(index => index.ToString(), index => new Person
+            {
+                Id = index.ToString(),
+                Name = $"Sussan Tables {index}",
+                Age = 40 + index,
+                Email = "sussan@example.com"
+            });
+
+        public static void UsePersonEndpoints(this WebApplication app)
+        {
+            app.MapGet("/persons", () => _persons.Values.ToArray())
+                .Produces<List<Person>>(200, "application/json")
+                .WithOpenApi(operation =>
+                {
+                    operation.OperationId = "ListPersons";
+                    operation.Description = "A list of persons.";
+                    operation.Summary = "A list of persons.";
+                    operation.Responses["200"].Description = "A list of persons.";
+                    return operation;
+                });
+            app.MapPost("/persons", async (HttpContext http, IValidator<NewPerson> validator, [FromBody] NewPerson newPerson) =>
+                {
+                    FluentValidation.Results.ValidationResult results = await validator.ValidateAsync(newPerson);
```

Comment:
What's good about this is we can have multiple errors returned in a single validation pass, while throwing an exception would only allow for one error.